src/run/serve.rs

```diff
@@ -0,0 +1,60 @@
+//! Serving the app locally for the browser.
+use actix_web::{rt, web, App, HttpResponse, HttpServer, Responder};
+use std::path::Path;
+
+use crate::external_cli::wasm_bindgen;
+
+/// If the user didn't provide an `index.html`, serve a default one.
+async fn serve_default_index() -> impl Responder {
+    let content = include_str!(concat!(
+        env!("CARGO_MANIFEST_DIR"),
+        "/assets/web/index.html"
+    ));
+
+    // Build the HTTP response with appropriate headers to serve the content as a file
+    HttpResponse::Ok()
+        .insert_header((
+            actix_web::http::header::CONTENT_TYPE,
+            "text/html; charset=utf-8",
+        ))
+        .body(content)
+}
+
+/// Launch a web server running the Bevy app.
+pub(crate) fn serve(port: u16, profile: &str) -> anyhow::Result<()> {
+    let profile = profile.to_string();
+
+    rt::System::new().block_on(
+        HttpServer::new(move || {
+            let mut app = App::new();
+
+            // Serve the build artifacts at the `/build/*` route
+            // A custom `index.html` will have to call `/build/bevy_app.js`
+            let js_path = Path::new("bevy_app.js");
+            let wasm_path = Path::new("bevy_app_bg.wasm");
```

Comment:
Where do these magic values come from?

Reply:
They are the files generated by `wasm-bindgen`. I moved the filter function there to make this clearer :)